### ISSUE TITLE
Rules should raise argument error when a "base rule" is not provided

### ISSUE THREAD

**Issue by backus** [dry-rb/dry-validation#210]
Transcript from gitter:

> **John Backus @backus Jul 10 21:46**
> 
> ``` ruby
> require 'bundler/setup'
> 
> gem 'dry-validation', '~> 0.9.0'
> 
> require 'dry/validation'
> require 'dry/validation/version'
> 
> Dry::Validation::VERSION # => 0.9.0
> 
> SCHEMA =
>   Dry::Validation.Schema do
>     required(:bar).maybe(:str?)
> 
>     rule(conditional_foo: %i[foo bar]) do |foo, bar|
>       foo.size?(9) | bar.str?
>     end
>   end
> 
> SCHEMA.call(foo: '123456789', bar: nil).messages # => {}
> SCHEMA.call(foo: '12345678', bar: nil).messages  # => {}
> SCHEMA.call(foo: nil, bar: nil).messages         # => {}
> ```
> 
> **John Backus @backus Jul 10 23:02**
> ah got it working by doing this:
> 
> ``` ruby
> SCHEMA =
>   Dry::Validation.Schema do
>     required(:foo).maybe(:str?)
>     required(:bar).maybe(:str?)
> 
>     rule(conditional_foo: %i[foo bar]) do |foo, bar|
>       (foo.str? & foo.size?(9)) | bar.str?
>     end
>   end
> ```
> 
> **Piotr Solnica @solnic 02:53**
> @backus pls report an issue, would be nice to raise an arg error when you try to do `foo.str?` when there’s no “base” rule for `:foo`

Link to this gitter discussion: https://gitter.im/dry-rb/chat?at=578324bcb79455146f8ca476

**Comment by flash-gordon** [comment]
Rules, as we know them, will be gone in the next version ☠️
